File: Python/Insertion_Selection_Sort.py
# Selection Sort
def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]

# Insertion Sort
def insertion_sort(arr):
    n = len(arr)
    for i in range(1, n):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key

# Main program
arr = [64, 25, 12, 22, 11]
choice = int(input("Choose Sorting Method:\n1. Selection Sort\n2. Insertion Sort\n"))

if choice == 1:
    selection_sort(arr)
    print("Sorted array using Selection Sort:", arr)
elif choice == 2:
    insertion_sort(arr)
    print("Sorted array using Insertion Sort:", arr)
else:
    print("Invalid choice")

File: C/Insertion_Selection_Sort.c
#include <stdio.h>

// Selection Sort
void selectionSort(int arr[], int n) {
	int i,j;
    for (i = 0; i < n - 1; i++) {
        int minIndex = i;
        for (j = i + 1; j < n; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }
        // Swap
        int temp = arr[minIndex];
        arr[minIndex] = arr[i];
        arr[i] = temp;
    }
}

// Insertion Sort
void insertionSort(int arr[], int n) {
	int i;
    for (i = 1; i < n; i++) {
        int key = arr[i];
        int j = i - 1;
        // Shift elements to the right
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
}

// Function to print array
void printArray(int arr[], int n) {
	int i;
    for (i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main() {
    int arr[] = {64, 25, 12, 22, 11};
    int n = sizeof(arr) / sizeof(arr[0]);
    int choice;

    printf("Choose Sorting Method:\n");
    printf("1. Selection Sort\n");
    printf("2. Insertion Sort\n");
    scanf("%d", &choice);

    if (choice == 1) {
        selectionSort(arr, n);
        printf("Sorted array using Selection Sort: ");
    } 
    else if (choice == 2) {
        insertionSort(arr, n);
        printf("Sorted array using Insertion Sort: ");
    } 
    else {
        printf("Invalid choice\n");
        return 0;
    }

    printArray(arr, n);
    return 0;
}

